docs/pages/eas/workflows/pre-packaged-jobs.mdx

```diff
@@ -539,6 +537,123 @@ jobs:
+
+```yaml
+jobs:
+  testflight_distribution:
+    type: testflight
+    params:
+      build_id: string # required
+      profile: string # optional, default: production
+      internal_groups: string[] # optional
+      external_groups: string[] # optional
+      changelog: string # optional
+      submit_beta_review: boolean # optional
+```
+
+#### Parameters
+
+You can pass the following parameters into the `params` list:
+
+| Parameter          | Type     | Description                                                                                                                                 |
+| ------------------ | -------- | ------------------------------------------------------------------------------------------------------------------------------------------- |
+| build_id           | string   | Required. The ID of the iOS build to distribute.                                                                                            |
+| profile            | string   | Optional. The submit profile to use. Defaults to `production`.                                                                              |
+| internal_groups    | string[] | Optional. An array of TestFlight internal group names to add the build to. Only include groups without automatic distribution enabled.      |
+| external_groups    | string[] | Optional. An array of TestFlight external group names to add the build to.                                                                  |
+| changelog          | string   | Optional. Test notes (what to test) for TestFlight testers.                                                                                 |
+| submit_beta_review | boolean  | Optional. Whether to submit for Beta App Review. If not specified, defaults to `true` when external_groups are provided, `false` otherwise. |
+
+#### Outputs
+
+You can reference the following outputs in subsequent jobs:
+
+| Output                | Type   | Description                                       |
+| --------------------- | ------ | ------------------------------------------------- |
+| apple_app_id          | string | The Apple App ID of the submitted build.          |
+| ios_bundle_identifier | string | The iOS bundle identifier of the submitted build. |
+
+### Examples
+
+Here are some practical examples of using the TestFlight job:
+
+<Collapsible summary="Full distribution with internal and external groups">
+
+This workflow distributes to both internal and external TestFlight groups with changelog.
+
+```yaml .eas/workflows/testflight-full.yml
+name: TestFlight Distribution
+
+jobs:
+  build_ios:
+    name: Build iOS
+    type: build
+    params:
+      platform: ios
+      profile: production
+
+  testflight:
+    name: Distribute to TestFlight
+    type: testflight
+    needs: [build_ios]
+    params:
+      build_id: ${{ needs.build_ios.outputs.build_id }}
+      internal_groups: ['QA Team']
+      external_groups: ['Public Beta']
+      changelog: |
```

Comment:
I think it may be nice to show an example of `${{ github.commit_message || "Bug fixes" }}` or `${{ github.event.head_commit.message || "Bug fixes" }}`, maybe with `# only available on push events which this is` or sth